src/VecSim/spaces/L2/L2_SVE2_INT8.h

```diff
@@ -0,0 +1,105 @@
+/*
+ *Copyright Redis Ltd. 2021 - present
+ *Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ *the Server Side Public License v1 (SSPLv1).
+ */
+
+#include "VecSim/spaces/space_includes.h"
+#include <arm_sve.h>
+
+// Aligned step using svptrue_b8()
+inline void L2SquareStep(const int8_t *&pVect1, const int8_t *&pVect2, size_t &offset,
+                         svint32_t &sum, const size_t chunk) {
+    svbool_t pg = svptrue_b8();
+    // Note: Because all the bits are 1, the extention to 16 and 32 bits does not make a difference
+    // Otherwise, pg should be recalculated for 16 and 32 operations
+
+    svint8_t v1_i8 = svld1_s8(pg, pVect1 + offset); // Load int8 vectors from pVect1
+    svint8_t v2_i8 = svld1_s8(pg, pVect2 + offset); // Load int8 vectors from pVect2
+
+    // Subtract v2 from v1 and widen the results to int16 for the even indexes
+    svint16_t diff_e = svsublb_s16(v1_i8, v2_i8);
+
+    // Subtract v2 from v1 and widen the results to int16 for the odd indexes
+    svint16_t diff_o = svsublt_s16(v1_i8, v2_i8);
+
+    sum = svmlalb_s32(sum, diff_e, diff_e);
+    sum = svmlalt_s32(sum, diff_e, diff_e);
+
+    sum = svmlalb_s32(sum, diff_o, diff_o);
+    sum = svmlalt_s32(sum, diff_o, diff_o);
```

Comment:
Are there no intrinsics like `svdot_s32` to multiply and fold i16 into i32?

Reply:
svdot is SVE, can be used only for  i8 -> i32
There is an i16 -> i64,  but I doubt it will be faster due to needing type conversion/ accumulating on i64, I suggest to keep it as  a TODO.